packages/ui-react/src/containers/ShelfList/ShelfList.tsx

```diff
@@ -1,88 +1,136 @@
+
+        return autoRefetch ? 1000 * 30 : false;
+      },
+      retry: false,
+    })),
+  );
+
   useEffect(() => {
     // reset row count when the page changes
-    return () => setRowCount(INITIAL_ROW_COUNT);
+    return () => setRowsLoaded(INITIAL_ROWS_LOADED);
   }, [rows]);
 
+  // If all playlists are empty, due to geo restrictions, we show a geo block error page
+  const allPlaylistsEmpty = playlists.every(({ data, isSuccess }) => isSuccess && !data?.playlist?.length);
+
+  if (allPlaylistsEmpty) {
+    return <ErrorPage title={t('geo_blocked_heading')} message={t('geo_blocked_description')} />;
+  }
+
   return (
-    <div className={styles.home}>
+    <div className={styles.shelfList}>
       <InfiniteScroll
-        pageStart={0}
         style={{ overflow: 'hidden' }}
-        loadMore={() => setRowCount((current) => current + LOAD_ROWS_COUNT)}
-        hasMore={rowCount < rows.length}
+        loadMore={() => setRowsLoaded((current) => current + LOAD_ROWS_COUNT)}
+        hasMore={rowsLoaded < rows.length}
         role="grid"
         loader={<InfiniteScrollLoader key="loader" />}
       >
-        {rows.slice(0, rowCount).map((row, index) => (
-          <PlaylistContainer type={row.type} playlistId={row.contentId} key={`${row.contentId || row.type}_${index}`}>
-            {({ playlist, error, isLoading, style }) => {
-              const title = row?.title || playlist.title;
-              const posterAspect = parseAspectRatio(playlist.cardImageAspectRatio || playlist.shelfImageAspectRatio);
-              const visibleTilesDelta = parseTilesDelta(posterAspect);
-
-              return (
-                <div
-                  style={style}
-                  role="row"
-                  className={classNames(styles.shelfContainer, { [styles.featured]: row.featured })}
-                  data-testid={testId(`shelf-${row.featured ? 'featured' : row.type !== 'playlist' ? row.type : slugify(title)}`)}
-                >
-                  <div role="cell">
-                    <ShelfComponent
-                      loading={isLoading}
-                      error={error}
-                      type={row.type}
-                      playlist={playlist}
-                      watchHistory={row.type === PersonalShelf.ContinueWatching ? watchHistoryDictionary : undefined}
-                      title={title}
-                      featured={row.featured === true}
-                      accessModel={accessModel}
-                      isLoggedIn={!!user}
-                      hasSubscription={!!subscription}
-                      posterAspect={posterAspect}
-                      visibleTilesDelta={visibleTilesDelta}
-                    />
-                  </div>
-                </div>
-              );
-            }}
-          </PlaylistContainer>
-        ))}
+        {rows.slice(0, rowsLoaded - 1).map(({ type, featured, title }, index) => {
```

Comment:
As discussed on Slack but as a reminder: we want to show the amount of loaded rows. When we add `- 1` here, we will not show the last rendered row and therefore miss 1 row. We can remove the `-1` here!